dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java

```diff
@@ -242,7 +244,25 @@ else if ( !UtilMethods.isSet(type) && UtilMethods.isSet( folder = APILocator.get
                                           continue;
                                       }
                                       type = PusheableAsset.FOLDER.getType();
-                                  }
+                                  }  // check if it is an Experiment
+								  else if ( !UtilMethods.isSet(type) && APILocator.getExperimentsAPI().find(identifier, user).isPresent() ) {
```

Comment:
The operation of finding the experiment is running and probably getting the same result. Please cache the returning `Optional`.

Reply:
@dsilvam BTW, I'm noticed some cache commented code at `ExperimentFactoryImpl` class at `find(id)` method. 
Do you plan to enable cache in a further PR?